src/withRTLExtension.js

```diff
@@ -14,13 +16,9 @@ function directionSelectorHandler(selector, baseSelector, generateSubtreeStyles)
 
   const generated = generateSubtreeStyles(baseSelector);
 
-  // generated may include more than one style definition. We want to prefix each one individually.
-  // This primarily happens for pseudo selectors for which generated looks like
-  //   .classname::selector{...}.classname::otherselector{...}
-  // This is the only case we really worry about, and as a result, we split on '}.' which marks the
-  // end of style definition and beginning of a new one.
-  const styleDefs = generated.split('}.').map(g => `[dir="rtl"] ${g}`);
-  return styleDefs.join('}.');
+  // generated may include more than one style definition (pseudoselectors, matchmedia, etc.).
+  // We want to prefix each one individually.
+  return generated.replace(styleDefRegex, g => `[dir="rtl"] ${g}`);
```

Comment:
You should be able to do this replace without the function and just a capture group and back reference, right?

```js
const styleDefRegex = /(\.[^{}]+\{[^{}]+\})/g
...
return generated.replace(styleDefRegex, '[dir="rtl"] $1');
```

Reply:
Is that better?

Reply:
I would expect it to be faster, but I haven't tested.